### PR TITLE
✨ Maintenance/add consistency scripts

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- adds a script for checking consistency between the file entries in the projects/file_meta_data tables in the database and what is available on S3

The scripts does the following:
pre-requisite:
- local docker volume containing a database from a deployment (see ```make import-db-from-docker-volume``` in /packages/postgres-database)

1. from the local DB parses the projects table for all the node UUIDs available used by some non-GUEST user
2. from the local DB parses the file_meta_data table for all file entries corresponding to the node UUIDs from 1.
3. from the corresponding S3 endpoint parses all the file corresponding to the node UUIDs from 1.
4. Compare files parsed in DB to the ones in S3
5. create a report as csv file containing

### CODE REVIEW

File: scripts/maintenance/check_consistency_data.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        f"now connecting with S3 backend and getting file for {len(project_nodes)} projects..."
+    )
+    s3_file_entries = set()
+    with typer.progressbar(project_nodes) as progress:
+        for project_uuid in progress:
+            try:
+                completed_process = subprocess.run(
+                    f"docker run "
+                    f"--env MC_HOST_mys3='https://{s3_access}:{s3_secret}@{s3_endpoint}' "
+                    "minio/mc "
```
Comment (pcrespov) [inline scripts/maintenance/check_consistency_data.py:113]: for subprocess, why don't you pass arguments? see e.g. [this example](https://github.com/ITISFoundation/osparc-simcore/blob/32f6501d7336931f6458c4e0cadbabdab683f0db/packages/pytest-simcore/src/pytest_simcore/helpers/utils_docker.py#L1410)

File: scripts/maintenance/check_consistency_data.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    typer.echo(
+        f"now connecting with S3 backend and getting file for {len(project_nodes)} projects..."
+    )
+    s3_file_entries = set()
+    with typer.progressbar(project_nodes) as progress:
+        for project_uuid in progress:
+            try:
+                completed_process = subprocess.run(
+                    f"docker run "
+                    f"--env MC_HOST_mys3='https://{s3_access}:{s3_secret}@{s3_endpoint}' "
```
Comment (pcrespov) [inline scripts/maintenance/check_consistency_data.py:112]: so you do use the client installed in the image instead of installing it

Comment (sanderegg) [reply scripts/maintenance/check_consistency_data.py:112]: that is the way that minio documents how to access the server

**Comment by pcrespov** [comment]
A couple of suggestion:
- dump data in csv files with headers (can be opened and explored in excel-like)
- in ``s3_missing_files.csv`` add columns with file size and last modification date (see below). This helps to explore the consistency
- option to use dumped files as cache instead of re-running requests

image

File: scripts/maintenance/check_consistency_data.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    except subprocess.CalledProcessError:
+        pass
+
+    progress.update(1)
+    return s3_file_entries
+
+
+def write_file(file_path: Path, data, fieldnames):
+    with file_path.open("w", newline="") as csvfile:
+        csv_writer = csv.writer(csvfile)
```
Comment (pcrespov) [inline scripts/maintenance/check_consistency_data.py:202]: nce touch

**Review by pcrespov — APPROVED** [review]

**Review by mguidon — APPROVED** [review]
real nice

**Pull request merged**